# Livro para consulta:
- https://jakevdp.github.io/PythonDataScienceHandbook/03.08-aggregation-and-grouping.html
- https://jakevdp.github.io/PythonDataScienceHandbook/03.09-pivot-tables.html
    

# 1. Importando bibliotecas <a name="import"></a>

<div style="text-align: right"
     
[Voltar ao índice](#Contents)

In [1]:
import pandas as pd

# 2. Carregando o dataframe SINASC <a name="read"></a>
<div style="text-align: right"
     
[Voltar ao índice](#Contents)

In [2]:
sinasc_raw = pd.read_csv('../SINASC_RO_2019.csv')
sinasc_raw.head()

ORIGEM   CODESTAB  CODMUNNASC  LOCNASC  IDADEMAE  ESTCIVMAE  \
0       1  2679477.0      110001        1        19        5.0   
1       1  2679477.0      110001        1        29        2.0   
2       1  2679477.0      110001        1        37        9.0   
3       1  2516500.0      110001        1        30        5.0   
4       1  2516500.0      110001        1        30        2.0   

            ESCMAE  CODOCUPMAE  QTDFILVIVO  QTDFILMORT  ...  KOTELCHUCK  \
0      8 a 11 anos         NaN         0.0         0.0  ...           5   
1      8 a 11 anos    999992.0         1.0         0.0  ...           5   
2      8 a 11 anos    513205.0         2.0         0.0  ...           5   
3  12 anos ou mais    231205.0         0.0         0.0  ...           4   
4      8 a 11 anos    999992.0         1.0         0.0  ...           5   

  CONTADOR munResStatus munResTipo               munResNome  munResUf  \
0        1        ATIVO      MUNIC    Alta Floresta D'Oeste  Rondônia   
1        2        ATIVO      MUNIC    Alta Floresta D'Oeste  Rondônia   
2        3        ATIVO      MUNIC    Alta Floresta D'Oeste  Rondônia   
3        4        ATIVO      MUNIC  Alto Alegre dos Parecis  Rondônia   
4        5        ATIVO      MUNIC    Alta Floresta D'Oeste  Rondônia   

   munResLat munResLon  munResAlt  munResArea  
0  -11.93554 -61.99982      338.0    7067.025  
1  -11.93554 -61.99982      338.0    7067.025  
2  -11.93554 -61.99982      338.0    7067.025  
3  -12.13178 -61.85308      397.0    3958.273  
4  -11.93554 -61.99982      338.0    7067.025  

[5 rows x 69 columns]

# Tarefa 1

### 1. Idade media das mães e dos pais por município (coluna munResNome)


In [3]:
sinasc_raw.columns

Index(['ORIGEM', 'CODESTAB', 'CODMUNNASC', 'LOCNASC', 'IDADEMAE', 'ESTCIVMAE',
       'ESCMAE', 'CODOCUPMAE', 'QTDFILVIVO', 'QTDFILMORT', 'CODMUNRES',
       'GESTACAO', 'GRAVIDEZ', 'PARTO', 'CONSULTAS', 'DTNASC', 'HORANASC',
       'SEXO', 'APGAR1', 'APGAR5', 'RACACOR', 'PESO', 'IDANOMAL', 'DTCADASTRO',
       'CODANOMAL', 'NUMEROLOTE', 'VERSAOSIST', 'DTRECEBIM', 'DIFDATA',
       'DTRECORIGA', 'NATURALMAE', 'CODMUNNATU', 'CODUFNATU', 'ESCMAE2010',
       'SERIESCMAE', 'DTNASCMAE', 'RACACORMAE', 'QTDGESTANT', 'QTDPARTNOR',
       'QTDPARTCES', 'IDADEPAI', 'DTULTMENST', 'SEMAGESTAC', 'TPMETESTIM',
       'CONSPRENAT', 'MESPRENAT', 'TPAPRESENT', 'STTRABPART', 'STCESPARTO',
       'TPNASCASSI', 'TPFUNCRESP', 'TPDOCRESP', 'DTDECLARAC', 'ESCMAEAGR1',
       'STDNEPIDEM', 'STDNNOVA', 'CODPAISRES', 'TPROBSON', 'PARIDADE',
       'KOTELCHUCK', 'CONTADOR', 'munResStatus', 'munResTipo', 'munResNome',
       'munResUf', 'munResLat', 'munResLon', 'munResAlt', 'munResArea'],
      dtype='object')

In [4]:
# Criando um novo df apenas com as colunas necessárias

idade_pais = sinasc_raw[['IDADEMAE', 'IDADEPAI', 'munResNome']]
idade_pais

IDADEMAE  IDADEPAI               munResNome
0            19      26.0    Alta Floresta D'Oeste
1            29      24.0    Alta Floresta D'Oeste
2            37      32.0    Alta Floresta D'Oeste
3            30      24.0  Alto Alegre dos Parecis
4            30      27.0    Alta Floresta D'Oeste
...         ...       ...                      ...
27023        32      35.0                  Vilhena
27024        19       NaN              Chupinguaia
27025        24      30.0                  Vilhena
27026        21      24.0                  Vilhena
27027        18       NaN  Novo Horizonte do Oeste

[27028 rows x 3 columns]

In [5]:
idade_pais.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 27028 entries, 0 to 27027
Data columns (total 3 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   IDADEMAE    27028 non-null  int64  
 1   IDADEPAI    7607 non-null   float64
 2   munResNome  27028 non-null  object 
dtypes: float64(1), int64(1), object(1)
memory usage: 633.6+ KB


In [6]:
idade_media_pais_mun = idade_pais.groupby('munResNome').mean()
idade_media_pais_mun

IDADEMAE   IDADEPAI
munResNome                                     
Alta Floresta D'Oeste      25.991826  29.548701
Alto Alegre dos Parecis    24.844156  29.179104
Alto Paraíso               24.960177  28.833333
Alvorada D'Oeste           25.770732  30.757282
Ariquemes                  25.607866  32.466667
Buritis                    25.559177  30.812500
Cabixi                     26.037500  34.333333
Cacaulândia                25.546667  36.333333
Cacoal                     26.890102  30.931854
Campo Novo de Rondônia     24.825301  30.500000
Candeias do Jamari         25.243323  29.894737
Castanheiras               27.281250  30.392857
Cerejeiras                 27.240506  31.652778
Chupinguaia                25.333333  29.629630
Colorado do Oeste          27.602804  34.354839
Corumbiara                 24.716535  32.470588
Costa Marques              24.412017  30.613636
Cujubim                    24.482927  31.400000
Espigão D'Oeste            26.093333  31.198113
Governador Jorge Teixeira  24.401786  29.222222
Guajará-Mirim              25.462228  30.470588
Itapuã do Oeste            26.064220  42.500000
Jaru                       25.982974  32.197183
Ji-Paraná                  26.361137  31.132239
Machadinho D'Oeste         25.480000  31.326923
Ministro Andreazza         27.904762  33.849315
Mirante da Serra           26.304878  32.154762
Monte Negro                25.648241  34.678571
Município ignorado - RO    24.000000  22.000000
Nova Brasilândia D'Oeste   26.222222  30.520833
Nova Mamoré                25.326389  31.692308
Nova União                 26.894231  31.371429
Novo Horizonte do Oeste    26.938776  31.588235
Ouro Preto do Oeste        26.375594  31.233244
Parecis                    26.454545  32.473684
Pimenta Bueno              25.199675  29.745174
Pimenteiras do Oeste       25.075000  35.600000
Porto Velho                26.312670  32.352679
Presidente Médici          25.796053  30.150754
Primavera de Rondônia      27.023256  31.888889
Rio Crespo                 26.660000  28.000000
Rolim de Moura             26.600863  30.360479
Santa Luzia D'Oeste        26.121212  30.703297
Seringueiras               24.910714  30.500000
São Felipe D'Oeste         26.481481  31.395833
São Francisco do Guaporé   25.660066  31.638889
São Miguel do Guaporé      25.926346  32.666667
Teixeirópolis              26.031250  31.500000
Theobroma                  25.536082  30.083333
Urupá                      26.688235  31.008929
Vale do Anari              26.096774  34.000000
Vale do Paraíso            25.679012  29.976744
Vilhena                    26.433962  34.152941

Acima é possível ver um data frame contendo a idade média das mães e dos pais em cada cidade em nossa base de dados. Porém, anteriormente vimos que existem dados faltantes, especificamente a idade dos pais. Abaixo será criado um novo data frame contendo a mesma quantidade de pais e mães, serão excluídas as linhas com dados faltantes, trazendo um novo resultado de idade média das mães, que pode ser melhor para fazer comparação entre as médias das idades.

In [7]:
idade_pais_sem_na = idade_pais.dropna()
idade_pais_sem_na.info()

<class 'pandas.core.frame.DataFrame'>
Index: 7607 entries, 0 to 27026
Data columns (total 3 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   IDADEMAE    7607 non-null   int64  
 1   IDADEPAI    7607 non-null   float64
 2   munResNome  7607 non-null   object 
dtypes: float64(1), int64(1), object(1)
memory usage: 237.7+ KB


In [8]:
idade_media_pais_mun_sem_na = idade_pais_sem_na.pivot_table(index = 'munResNome') # por padrão, o pivot table já agrupa o index pela média
idade_media_pais_mun_sem_na

IDADEMAE   IDADEPAI
munResNome                                     
Alta Floresta D'Oeste      25.892857  29.548701
Alto Alegre dos Parecis    24.679104  29.179104
Alto Paraíso               22.833333  28.833333
Alvorada D'Oeste           26.825243  30.757282
Ariquemes                  28.133333  32.466667
Buritis                    24.750000  30.812500
Cabixi                     32.166667  34.333333
Cacaulândia                31.000000  36.333333
Cacoal                     26.978605  30.931854
Campo Novo de Rondônia     24.900000  30.500000
Candeias do Jamari         23.684211  29.894737
Castanheiras               28.071429  30.392857
Cerejeiras                 28.361111  31.652778
Chupinguaia                26.888889  29.629630
Colorado do Oeste          31.161290  34.354839
Corumbiara                 29.000000  32.470588
Costa Marques              25.250000  30.613636
Cujubim                    27.400000  31.400000
Espigão D'Oeste            26.808176  31.198113
Governador Jorge Teixeira  25.555556  29.222222
Guajará-Mirim              26.058824  30.470588
Itapuã do Oeste            31.375000  42.500000
Jaru                       28.450704  32.197183
Ji-Paraná                  27.282819  31.132239
Machadinho D'Oeste         26.048077  31.326923
Ministro Andreazza         27.890411  33.849315
Mirante da Serra           26.928571  32.154762
Monte Negro                27.642857  34.678571
Município ignorado - RO    24.000000  22.000000
Nova Brasilândia D'Oeste   27.104167  30.520833
Nova Mamoré                26.051282  31.692308
Nova União                 27.371429  31.371429
Novo Horizonte do Oeste    26.988235  31.588235
Ouro Preto do Oeste        27.109920  31.233244
Parecis                    26.973684  32.473684
Pimenta Bueno              25.449807  29.745174
Pimenteiras do Oeste       32.600000  35.600000
Porto Velho                28.355655  32.352679
Presidente Médici          26.170854  30.150754
Primavera de Rondônia      28.083333  31.888889
Rio Crespo                 26.000000  28.000000
Rolim de Moura             26.549701  30.360479
Santa Luzia D'Oeste        26.043956  30.703297
Seringueiras               25.058824  30.500000
São Felipe D'Oeste         26.979167  31.395833
São Francisco do Guaporé   26.944444  31.638889
São Miguel do Guaporé      27.469136  32.666667
Teixeirópolis              27.454545  31.500000
Theobroma                  24.916667  30.083333
Urupá                      27.116071  31.008929
Vale do Anari              27.133333  34.000000
Vale do Paraíso            25.604651  29.976744
Vilhena                    30.611765  34.152941

### 2. Peso médio dos bebes por sexo que nasceram no dia do seu aniversário por faixas de escolaridade mae
Ex: Você, aluna(o), nasceu no dia 10/01, então você precisa filtrar o conjunto de dados nessa data e calcular o peso médio dos bebês de cada sexo por faixa de escolaridade da mãe.

In [9]:
sinasc_raw.columns

Index(['ORIGEM', 'CODESTAB', 'CODMUNNASC', 'LOCNASC', 'IDADEMAE', 'ESTCIVMAE',
       'ESCMAE', 'CODOCUPMAE', 'QTDFILVIVO', 'QTDFILMORT', 'CODMUNRES',
       'GESTACAO', 'GRAVIDEZ', 'PARTO', 'CONSULTAS', 'DTNASC', 'HORANASC',
       'SEXO', 'APGAR1', 'APGAR5', 'RACACOR', 'PESO', 'IDANOMAL', 'DTCADASTRO',
       'CODANOMAL', 'NUMEROLOTE', 'VERSAOSIST', 'DTRECEBIM', 'DIFDATA',
       'DTRECORIGA', 'NATURALMAE', 'CODMUNNATU', 'CODUFNATU', 'ESCMAE2010',
       'SERIESCMAE', 'DTNASCMAE', 'RACACORMAE', 'QTDGESTANT', 'QTDPARTNOR',
       'QTDPARTCES', 'IDADEPAI', 'DTULTMENST', 'SEMAGESTAC', 'TPMETESTIM',
       'CONSPRENAT', 'MESPRENAT', 'TPAPRESENT', 'STTRABPART', 'STCESPARTO',
       'TPNASCASSI', 'TPFUNCRESP', 'TPDOCRESP', 'DTDECLARAC', 'ESCMAEAGR1',
       'STDNEPIDEM', 'STDNNOVA', 'CODPAISRES', 'TPROBSON', 'PARIDADE',
       'KOTELCHUCK', 'CONTADOR', 'munResStatus', 'munResTipo', 'munResNome',
       'munResUf', 'munResLat', 'munResLon', 'munResAlt', 'munResArea'],
      dtype='object')

In [10]:
peso_bebe = sinasc_raw[['ESCMAE', 'DTNASC', 'SEXO', 'PESO']]
peso_bebe

ESCMAE      DTNASC       SEXO  PESO
0          8 a 11 anos  2019-02-19  Masculino  3685
1          8 a 11 anos  2019-02-21   Feminino  3055
2          8 a 11 anos  2019-02-25   Feminino  3460
3      12 anos ou mais  2019-03-20   Feminino  2982
4          8 a 11 anos  2019-03-23   Feminino  3310
...                ...         ...        ...   ...
27023  12 anos ou mais  2019-12-13  Masculino  2110
27024      8 a 11 anos  2019-10-04  Masculino  3310
27025      8 a 11 anos  2019-08-02  Masculino  3065
27026      8 a 11 anos  2019-12-23  Masculino  2352
27027      8 a 11 anos  2019-12-27  Masculino  4240

[27028 rows x 4 columns]

In [11]:
peso_bebe[peso_bebe['DTNASC'] == '2019-09-05'].groupby(['ESCMAE', 'SEXO'])['PESO'].mean().to_frame()

PESO
ESCMAE          SEXO                  
1 a 3 anos      Feminino   3450.000000
12 anos ou mais Feminino   3449.166667
                Masculino  3192.500000
4 a 7 anos      Feminino   3063.750000
                Masculino  3249.166667
8 a 11 anos     Feminino   3107.857143
                Masculino  3269.821429

In [12]:
peso_bebe[peso_bebe['DTNASC'] == '2019-09-05'].count()

ESCMAE    69
DTNASC    71
SEXO      71
PESO      71
dtype: int64

### 3. Qual o municipio que nasceu menos bebe em 2019?
    - qual a idade media, maxima, minima das maes nesse municipio?
    - qual a idade media, maxima, minima dos pais nesse municipio?

In [13]:
sinasc_raw['munResNome'].value_counts(ascending=True).head(5)

munResNome
Município ignorado - RO     1
Castanheiras               32
Pimenteiras do Oeste       40
Primavera de Rondônia      43
Parecis                    44
Name: count, dtype: int64

In [14]:
sinasc_raw['munResNome'].value_counts(ascending=True).tail(5)

munResNome
Cacoal         1374
Vilhena        1590
Ariquemes      1729
Ji-Paraná      2182
Porto Velho    8437
Name: count, dtype: int64

In [15]:
sinasc_raw.shape

(27028, 69)

In [16]:
(32/27028)*100

0.11839573775344087

O município em que nasceu menos bebês foi o de Castanheiras, onde nasceram apenas 32 bebês. Isso equivale 20% a menos que a segunda cidade que menos nasceu bebês (Pimenteiras do Oeste, com 40) e 99,62% em comparação à cidade que mais nasceu (Porto Velho, com 8437). Em comparação a base toda, essa cidade representa 0,12% dos nascimentos em todo o estado de Rondônia em 2019. Ainda existe o registro de um município como ignorado.

In [17]:
castanheiras = sinasc_raw[sinasc_raw['munResNome'] == 'Castanheiras']

In [18]:
idade_maes_castanheiras = castanheiras.groupby('munResNome')['IDADEMAE'].agg(['mean', 'max', 'min'])
idade_maes_castanheiras

mean  max  min
munResNome                      
Castanheiras  27.28125   39   17

In [19]:
idade_pais_castanheiras = castanheiras.groupby('munResNome')['IDADEPAI'].agg(['mean', 'max', 'min'])
idade_pais_castanheiras

mean   max   min
munResNome                         
Castanheiras  30.392857  43.0  17.0

### 4. Qual o municipio que nasceu mais bebe no mês de março?
    - qual a quantidade de filhos vivos media, maxima, minima nesse municipio?
    - qual a idade media, maxima, minima dos pais nesse municipio?



In [20]:
sinasc_raw['DTNASC']

0        2019-02-19
1        2019-02-21
2        2019-02-25
3        2019-03-20
4        2019-03-23
            ...    
27023    2019-12-13
27024    2019-10-04
27025    2019-08-02
27026    2019-12-23
27027    2019-12-27
Name: DTNASC, Length: 27028, dtype: object

In [21]:
sinasc_raw['DATA NASCIMENTO'] = pd.to_datetime(sinasc_raw['DTNASC'], format='%Y-%m-%d')
sinasc_raw['DATA NASCIMENTO']

0       2019-02-19
1       2019-02-21
2       2019-02-25
3       2019-03-20
4       2019-03-23
           ...    
27023   2019-12-13
27024   2019-10-04
27025   2019-08-02
27026   2019-12-23
27027   2019-12-27
Name: DATA NASCIMENTO, Length: 27028, dtype: datetime64[ns]

In [22]:
marco = sinasc_raw[sinasc_raw['DATA NASCIMENTO'].dt.month == 3]
marco.head()

ORIGEM   CODESTAB  CODMUNNASC  LOCNASC  IDADEMAE  ESTCIVMAE  \
3        1  2516500.0      110001        1        30        5.0   
4        1  2516500.0      110001        1        30        2.0   
5        1  2516500.0      110001        1        33        5.0   
9        1  2679477.0      110001        1        30        2.0   
10       1  2679477.0      110001        1        31        5.0   

             ESCMAE  CODOCUPMAE  QTDFILVIVO  QTDFILMORT  ...  CONTADOR  \
3   12 anos ou mais    231205.0         0.0         0.0  ...         4   
4       8 a 11 anos    999992.0         1.0         0.0  ...         5   
5        1 a 3 anos    621005.0         2.0         1.0  ...         6   
9        1 a 3 anos    999992.0         1.0         1.0  ...        10   
10      8 a 11 anos    512105.0         1.0         0.0  ...        11   

   munResStatus munResTipo               munResNome  munResUf munResLat  \
3         ATIVO      MUNIC  Alto Alegre dos Parecis  Rondônia -12.13178   
4         ATIVO      MUNIC    Alta Floresta D'Oeste  Rondônia -11.93554   
5         ATIVO      MUNIC    Alta Floresta D'Oeste  Rondônia -11.93554   
9         ATIVO      MUNIC    Alta Floresta D'Oeste  Rondônia -11.93554   
10        ATIVO      MUNIC    Alta Floresta D'Oeste  Rondônia -11.93554   

    munResLon munResAlt  munResArea  DATA NASCIMENTO  
3   -61.85308     397.0    3958.273       2019-03-20  
4   -61.99982     338.0    7067.025       2019-03-23  
5   -61.99982     338.0    7067.025       2019-03-27  
9   -61.99982     338.0    7067.025       2019-03-11  
10  -61.99982     338.0    7067.025       2019-03-11  

[5 rows x 70 columns]

In [23]:
marco['munResNome'].value_counts().head()

munResNome
Porto Velho    744
Ji-Paraná      188
Vilhena        148
Ariquemes      141
Cacoal         133
Name: count, dtype: int64

In [24]:
marco['munResNome'].value_counts().sum()

2456

In [25]:
(744/2456)*100

30.293159609120522

A cidade em que mais nasceram bebês no mês de março foi Porto Velho com 744 bebês, isso representa um pouco mais que 30%.

In [26]:
porto_velho = sinasc_raw[sinasc_raw['munResNome'] == 'Porto Velho']
porto_velho.head()

ORIGEM   CODESTAB  CODMUNNASC  LOCNASC  IDADEMAE  ESTCIVMAE       ESCMAE  \
123       1  2515520.0      110002        1        22        1.0  8 a 11 anos   
201       1  2494299.0      110002        1        22        2.0  8 a 11 anos   
246       1  2494299.0      110002        1        29        1.0  8 a 11 anos   
272       1  2494299.0      110002        1        27        2.0   4 a 7 anos   
355       1  2515504.0      110002        1        33        2.0  8 a 11 anos   

     CODOCUPMAE  QTDFILVIVO  QTDFILMORT  ...  CONTADOR munResStatus  \
123    999992.0         0.0         0.0  ...       125        ATIVO   
201    999992.0         0.0         0.0  ...       204        ATIVO   
246    999992.0         2.0         0.0  ...       249        ATIVO   
272    999992.0         2.0         1.0  ...       275        ATIVO   
355    999992.0         0.0         0.0  ...       359        ATIVO   

    munResTipo   munResNome  munResUf munResLat  munResLon munResAlt  \
123      MUNIC  Porto Velho  Rondônia  -8.76889  -63.83145      87.0   
201      MUNIC  Porto Velho  Rondônia  -8.76889  -63.83145      87.0   
246      MUNIC  Porto Velho  Rondônia  -8.76889  -63.83145      87.0   
272      MUNIC  Porto Velho  Rondônia  -8.76889  -63.83145      87.0   
355      MUNIC  Porto Velho  Rondônia  -8.76889  -63.83145      87.0   

     munResArea  DATA NASCIMENTO  
123   34096.394       2019-01-11  
201   34096.394       2019-01-08  
246   34096.394       2019-01-23  
272   34096.394       2019-01-19  
355   34096.394       2019-02-08  

[5 rows x 70 columns]

In [27]:
porto_velho_marco = marco[marco['munResNome'] == 'Porto Velho']
porto_velho_marco.head()

ORIGEM   CODESTAB  CODMUNNASC  LOCNASC  IDADEMAE  ESTCIVMAE  \
612        1  2494299.0      110002        1        19        5.0   
700        1  2494299.0      110002        1        19        5.0   
3232       1  2808609.0      110011        1        34        5.0   
3238       1  2808609.0      110011        1        24        5.0   
6818       1  2515369.0      110020        1        38        2.0   

               ESCMAE  CODOCUPMAE  QTDFILVIVO  QTDFILMORT  ...  CONTADOR  \
612        4 a 7 anos    999992.0         0.0         0.0  ...       618   
700        4 a 7 anos    622020.0         1.0         0.0  ...       706   
3232          Nenhuma    622020.0         0.0         1.0  ...      3257   
3238      8 a 11 anos    622020.0         1.0         0.0  ...      3263   
6818  12 anos ou mais    999991.0         2.0         1.0  ...      6864   

     munResStatus munResTipo   munResNome  munResUf munResLat  munResLon  \
612         ATIVO      MUNIC  Porto Velho  Rondônia  -8.76889  -63.83145   
700         ATIVO      MUNIC  Porto Velho  Rondônia  -8.76889  -63.83145   
3232        ATIVO      MUNIC  Porto Velho  Rondônia  -8.76889  -63.83145   
3238        ATIVO      MUNIC  Porto Velho  Rondônia  -8.76889  -63.83145   
6818        ATIVO      MUNIC  Porto Velho  Rondônia  -8.76889  -63.83145   

     munResAlt  munResArea  DATA NASCIMENTO  
612       87.0   34096.394       2019-03-09  
700       87.0   34096.394       2019-03-28  
3232      87.0   34096.394       2019-03-11  
3238      87.0   34096.394       2019-03-13  
6818      87.0   34096.394       2019-03-01  

[5 rows x 70 columns]

In [28]:
qtde_filhos_vivos_porto_velho = porto_velho.groupby('munResNome')['QTDFILVIVO'].agg(['mean', 'max', 'min'])
qtde_filhos_vivos_porto_velho

mean   max  min
munResNome                      
Porto Velho  1.167236  12.0  0.0

In [29]:
qtde_filhos_vivos_porto_velho_marco = porto_velho_marco.groupby('munResNome')['QTDFILVIVO'].agg(['mean', 'max', 'min'])
qtde_filhos_vivos_porto_velho_marco

mean  max  min
munResNome                     
Porto Velho  1.295056  9.0  0.0

In [30]:
porto_velho['QTDFILVIVO'].value_counts()

QTDFILVIVO
1.0     2721
0.0     2411
2.0     1358
3.0      483
4.0      189
5.0       87
6.0       46
7.0       14
8.0        5
12.0       2
9.0        2
10.0       1
Name: count, dtype: int64

In [31]:
porto_velho['QTDFILVIVO'].value_counts().sum()

7319

Durante todo o ano, houve um total de 7319 nascimentos na cidade de Porto Velho. 4908 bebês vivos, sendo 0 a quantidade mínima, 12 o máximo e média é de 1,17.

In [32]:
porto_velho_marco['QTDFILVIVO'].value_counts()

QTDFILVIVO
1.0    241
0.0    175
2.0    133
3.0     39
4.0     19
5.0     12
6.0      6
7.0      1
9.0      1
Name: count, dtype: int64

In [33]:
porto_velho_marco['QTDFILVIVO'].value_counts().sum()

627

No mês de março, houve um total de 627 nascimentos na cidade de Porto Velho. 452 bebês vivos, sendo 0 a quantidade mínima, 9 o máximo e média é de 1,3.

In [34]:
idade_mae_porto_velho = porto_velho.groupby('munResNome')['IDADEMAE'].agg(['mean', 'max', 'min'])
idade_mae_porto_velho

mean  max  min
munResNome                     
Porto Velho  26.31267   47   12

In [35]:
idade_mae_porto_velho_marco = porto_velho_marco.groupby('munResNome')['IDADEMAE'].agg(['mean', 'max', 'min'])
idade_mae_porto_velho_marco

mean  max  min
munResNome                      
Porto Velho  26.596774   44   14

In [36]:
idade_pai_porto_velho = porto_velho.groupby('munResNome')['IDADEPAI'].agg(['mean', 'max', 'min'])
idade_pai_porto_velho

mean   max   min
munResNome                        
Porto Velho  32.352679  65.0  16.0

In [37]:
idade_pai_porto_velho_marco = porto_velho_marco.groupby('munResNome')['IDADEPAI'].agg(['mean', 'max', 'min'])
idade_pai_porto_velho_marco

mean   max   min
munResNome                        
Porto Velho  34.629032  62.0  19.0

### Analise as respostas encontradas, tire algum insight delas, conte pra gente algo encontrado nos dados. Algo que você julgue relevante e novo pra você.

Exemplo:
- Ah, descobri que a idade mediana das mulheres que deram a luz no ano de 2019 dos municipios x é maior que y.

In [38]:
sinasc_raw.dtypes

ORIGEM                      int64
CODESTAB                  float64
CODMUNNASC                  int64
LOCNASC                     int64
IDADEMAE                    int64
                        ...      
munResLat                 float64
munResLon                 float64
munResAlt                 float64
munResArea                float64
DATA NASCIMENTO    datetime64[ns]
Length: 70, dtype: object

In [39]:
sinasc_raw.select_dtypes(exclude=['int64','float64'])

ESCMAE         GESTACAO GRAVIDEZ    PARTO      DTNASC  \
0          8 a 11 anos  37 a 41 semanas    Única  Cesáreo  2019-02-19   
1          8 a 11 anos  37 a 41 semanas    Única  Cesáreo  2019-02-21   
2          8 a 11 anos  37 a 41 semanas    Única  Cesáreo  2019-02-25   
3      12 anos ou mais  37 a 41 semanas    Única  Cesáreo  2019-03-20   
4          8 a 11 anos  37 a 41 semanas    Única  Cesáreo  2019-03-23   
...                ...              ...      ...      ...         ...   
27023  12 anos ou mais  32 a 36 semanas    Única  Cesáreo  2019-12-13   
27024      8 a 11 anos  37 a 41 semanas    Única  Vaginal  2019-10-04   
27025      8 a 11 anos  37 a 41 semanas    Única  Cesáreo  2019-08-02   
27026      8 a 11 anos  32 a 36 semanas    Única  Vaginal  2019-12-23   
27027      8 a 11 anos  37 a 41 semanas    Única  Cesáreo  2019-12-27   

            SEXO RACACOR  DTCADASTRO CODANOMAL VERSAOSIST   DTRECEBIM  \
0      Masculino  Branca  2019-03-08       NaN     3.2.01  2019-03-08   
1       Feminino  Branca  2019-03-08       NaN     3.2.01  2019-03-08   
2       Feminino  Branca  2019-03-08       NaN     3.2.01  2019-03-08   
3       Feminino   Parda  2019-04-03       NaN     3.2.01  2019-04-09   
4       Feminino   Parda  2019-04-03       NaN     3.2.01  2019-04-09   
...          ...     ...         ...       ...        ...         ...   
27023  Masculino  Branca  2020-09-28       NaN     3.2.01  2020-10-13   
27024  Masculino  Branca  2019-10-16       NaN     3.2.01  2019-10-21   
27025  Masculino   Parda  2019-08-07       NaN     3.2.00  2019-08-07   
27026  Masculino   Parda  2020-01-07       NaN     3.2.00  2020-01-07   
27027  Masculino   Parda  2020-10-15       NaN     3.2.01  2020-10-16   

      munResStatus munResTipo               munResNome  munResUf  \
0            ATIVO      MUNIC    Alta Floresta D'Oeste  Rondônia   
1            ATIVO      MUNIC    Alta Floresta D'Oeste  Rondônia   
2            ATIVO      MUNIC    Alta Floresta D'Oeste  Rondônia   
3            ATIVO      MUNIC  Alto Alegre dos Parecis  Rondônia   
4            ATIVO      MUNIC    Alta Floresta D'Oeste  Rondônia   
...            ...        ...                      ...       ...   
27023        ATIVO      MUNIC                  Vilhena  Rondônia   
27024        ATIVO      MUNIC              Chupinguaia  Rondônia   
27025        ATIVO      MUNIC                  Vilhena  Rondônia   
27026        ATIVO      MUNIC                  Vilhena  Rondônia   
27027        ATIVO      MUNIC  Novo Horizonte do Oeste  Rondônia   

      DATA NASCIMENTO  
0          2019-02-19  
1          2019-02-21  
2          2019-02-25  
3          2019-03-20  
4          2019-03-23  
...               ...  
27023      2019-12-13  
27024      2019-10-04  
27025      2019-08-02  
27026      2019-12-23  
27027      2019-12-27  

[27028 rows x 16 columns]

In [40]:
sinasc_raw.columns

Index(['ORIGEM', 'CODESTAB', 'CODMUNNASC', 'LOCNASC', 'IDADEMAE', 'ESTCIVMAE',
       'ESCMAE', 'CODOCUPMAE', 'QTDFILVIVO', 'QTDFILMORT', 'CODMUNRES',
       'GESTACAO', 'GRAVIDEZ', 'PARTO', 'CONSULTAS', 'DTNASC', 'HORANASC',
       'SEXO', 'APGAR1', 'APGAR5', 'RACACOR', 'PESO', 'IDANOMAL', 'DTCADASTRO',
       'CODANOMAL', 'NUMEROLOTE', 'VERSAOSIST', 'DTRECEBIM', 'DIFDATA',
       'DTRECORIGA', 'NATURALMAE', 'CODMUNNATU', 'CODUFNATU', 'ESCMAE2010',
       'SERIESCMAE', 'DTNASCMAE', 'RACACORMAE', 'QTDGESTANT', 'QTDPARTNOR',
       'QTDPARTCES', 'IDADEPAI', 'DTULTMENST', 'SEMAGESTAC', 'TPMETESTIM',
       'CONSPRENAT', 'MESPRENAT', 'TPAPRESENT', 'STTRABPART', 'STCESPARTO',
       'TPNASCASSI', 'TPFUNCRESP', 'TPDOCRESP', 'DTDECLARAC', 'ESCMAEAGR1',
       'STDNEPIDEM', 'STDNNOVA', 'CODPAISRES', 'TPROBSON', 'PARIDADE',
       'KOTELCHUCK', 'CONTADOR', 'munResStatus', 'munResTipo', 'munResNome',
       'munResUf', 'munResLat', 'munResLon', 'munResAlt', 'munResArea',
       'DATA NASCIMENTO

In [41]:
# Adicionar coluna com o mês
sinasc_raw['MES'] = sinasc_raw['DATA NASCIMENTO'].dt.month

# Agrupar por mês e cidade, contando os nascimentos
nascimentos_por_cidade = sinasc_raw.groupby(['MES', 'munResNome']).size().reset_index(name='TOTAL_NASCIMENTOS')

# Para cada mês, encontrar a cidade com o maior número de nascimentos
cidade_mais_nascimentos = nascimentos_por_cidade.loc[nascimentos_por_cidade.groupby('MES')['TOTAL_NASCIMENTOS'].idxmax()]

cidade_mais_nascimentos

MES   munResNome  TOTAL_NASCIMENTOS
36     1  Porto Velho                724
88     2  Porto Velho                669
140    3  Porto Velho                744
192    4  Porto Velho                734
244    5  Porto Velho                723
296    6  Porto Velho                696
348    7  Porto Velho                696
400    8  Porto Velho                721
452    9  Porto Velho                720
504   10  Porto Velho                685
556   11  Porto Velho                671
608   12  Porto Velho                654

In [42]:
# Encontrar a segunda cidade com mais nascimentos por mês
segunda_cidade_mais_nascimentos = nascimentos_por_cidade.groupby('MES').nth(1).reset_index()
segunda_cidade_mais_nascimentos

index  MES               munResNome  TOTAL_NASCIMENTOS
0       1    1  Alto Alegre dos Parecis                 14
1      53    2  Alto Alegre dos Parecis                 21
2     105    3  Alto Alegre dos Parecis                 10
3     157    4  Alto Alegre dos Parecis                 15
4     209    5  Alto Alegre dos Parecis                 10
5     261    6  Alto Alegre dos Parecis                 17
6     313    7  Alto Alegre dos Parecis                  9
7     365    8  Alto Alegre dos Parecis                 15
8     417    9  Alto Alegre dos Parecis                 14
9     469   10  Alto Alegre dos Parecis                  9
10    521   11  Alto Alegre dos Parecis                  8
11    573   12  Alto Alegre dos Parecis                 12

In [43]:
# Encontrar a terceira cidade com mais nascimentos por mês
terceira_cidade_mais_nascimentos = nascimentos_por_cidade.groupby('MES').nth(2).reset_index()
terceira_cidade_mais_nascimentos

index  MES    munResNome  TOTAL_NASCIMENTOS
0       2    1  Alto Paraíso                 19
1      54    2  Alto Paraíso                 20
2     106    3  Alto Paraíso                 22
3     158    4  Alto Paraíso                 17
4     210    5  Alto Paraíso                 22
5     262    6  Alto Paraíso                 18
6     314    7  Alto Paraíso                 13
7     366    8  Alto Paraíso                 24
8     418    9  Alto Paraíso                 25
9     470   10  Alto Paraíso                 17
10    522   11  Alto Paraíso                 14
11    574   12  Alto Paraíso                 15

In [44]:
# Encontrar a quarta cidade com mais nascimentos por mês
quarta_cidade_mais_nascimentos = nascimentos_por_cidade.groupby('MES').nth(3).reset_index()
quarta_cidade_mais_nascimentos

index  MES        munResNome  TOTAL_NASCIMENTOS
0       3    1  Alvorada D'Oeste                 14
1      55    2  Alvorada D'Oeste                 14
2     107    3  Alvorada D'Oeste                 22
3     159    4  Alvorada D'Oeste                 13
4     211    5  Alvorada D'Oeste                 22
5     263    6  Alvorada D'Oeste                 12
6     315    7  Alvorada D'Oeste                 22
7     367    8  Alvorada D'Oeste                  9
8     419    9  Alvorada D'Oeste                 25
9     471   10  Alvorada D'Oeste                 19
10    523   11  Alvorada D'Oeste                 17
11    575   12  Alvorada D'Oeste                 16

Em todo o ano de 2019, durante todos os meses, a cidade com mais nascimento foi Porto Velho, em seguida vem Ji-Paraná, que é a segunda em todos os meses, com exceção de Outubro, onde é a terceira. A terceira posição por mês possui 3 cidades diferentes, e o mesmo acontece para a quarta posição.